src/starterbot/BaseManager.cpp

```diff
@@ -0,0 +1,156 @@
+#include "BaseManager.h"
+
+BaseManager::BaseManager(UnitManager& unitManager) :
+    m_unitManager(unitManager) {
+}
+
+Base& BaseManager::getClosestBase(bw::Unit& unit) {
```

Comment:
The reference type `bw::Unit&` is unnecessary since `bw::Unit` is a typedef to a pointer type `bw::UnitInterface*`. So just use `bw::Unit`.